### PR TITLE
Add final_snapshot_identifier argument

### PR DESCRIPTION
**Pull request by drama17**
## What
* Define `final_snapshot_identifier` parameter to have custom speaking name

## Why
* Need to have this variable defined in case if `skip_final_snapshot` is set to `false`

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -35,7 +35,7 @@ resource "aws_db_instance" "default" {
   backup_retention_period     = "${var.backup_retention_period}"
   backup_window               = "${var.backup_window}"
   tags                        = "${module.label.tags}"
-  final_snapshot_identifier   = "${var.final_snapshot_identifier}"
+  final_snapshot_identifier   = "${module.label.namespace}-${module.label.stage}-${module.label.name}-final-snapshot-${md5(timestamp())}"
```
Comment (comeanother) [inline main.tf]: Use `terraform-null-label` for the snapshot name.

Comment (comeanother) [reply main.tf]: Make `final_snapshot_identifier` user-defined.
Use condition like `${length(var.final_snapshot_identifier) > 0 ? var.final_snapshot_identifier : module.final_snapshot_label.id}` as a value

File: main.tf [hunk 2]
```diff
@@ -3,11 +3,19 @@ module "label" {
...
   attributes = "${var.attributes}"
   tags       = "${var.tags}"
 }
 
+module "final_snapshot_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.1"
+  namespace  = "${var.namespace}"
+  name       = "${var.name}"
+  stage      = "${var.stage}"
+  attributes = ["${concat(var.attributes, list("final", "snapshot", "${md5(timestamp())}"))}"]
```
Comment (comeanother) [inline main.tf]: Use `compact(concat())` for removing empty elements.

File: main.tf [hunk 3]
```diff
@@ -35,7 +43,7 @@ resource "aws_db_instance" "default" {
   backup_retention_period     = "${var.backup_retention_period}"
   backup_window               = "${var.backup_window}"
   tags                        = "${module.label.tags}"
-  final_snapshot_identifier   = "${var.final_snapshot_identifier}"
+  final_snapshot_identifier   = "${module.final_snapshot_label.id}"
```
Comment (comeanother) [inline main.tf]: See https://github.com/cloudposse/terraform-aws-rds/pull/7#discussion_r149078668

File: main.tf [hunk 4]
```diff
@@ -3,11 +3,19 @@ module "label" {
   namespace  = "${var.namespace}"
   name       = "${var.name}"
   stage      = "${var.stage}"
-  delimiter  = "${var.delimiter}"
```
Comment (comeanother) [inline main.tf]: We should be able to override delimiter by setting `delimiter` argument.
Do not drop this.

File: main.tf [hunk 5]
```diff
@@ -3,11 +3,19 @@ module "label" {
...
   attributes = "${var.attributes}"
   tags       = "${var.tags}"
 }
 
+module "final_snapshot_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.1"
+  namespace  = "${var.namespace}"
+  name       = "${var.name}"
+  stage      = "${var.stage}"
```
Comment (comeanother) [inline main.tf]: Add `delimiter`.

File: variables.tf [hunk 6]
```diff
@@ -158,7 +158,3 @@ variable "snapshot_identifier" {
   description = "Snapshot name e.g: rds:production-2015-06-26-06-05"
   default = ""
 }
-
-variable "final_snapshot_identifier" {
```
Comment (comeanother) [inline variables.tf]: Do not delete this variable.

File: main.tf (hunk 4 above)
Comment (drama17) [reply main.tf]: We don't need to define it - default value is OK for us

File: variables.tf (hunk 6 above)
Comment (drama17) [reply variables.tf]: Why?

**Comment by drama17** [comment]
Terraform plan when variable `final_snapshot_identifier` is set


Terraform plan when `final_snapshot_identifier` is not set

**Pull request merged**